Video codec won't work properly with a clock too low nor
too high. We need to export them, allowing the device
tree to assign a suitable clocks for them.

Signed-off-by: Randy Li <ayaka@soulik.info>
---
 drivers/clk/rockchip/clk-rk3399.c      | 5 +++--
 include/dt-bindings/clock/rk3399-cru.h | 2 ++
 2 files changed, 5 insertions(+), 2 deletions(-)

diff --git a/drivers/clk/rockchip/clk-rk3399.c b/drivers/clk/rockchip/clk-rk3399.c
index 5a628148f3f0..fe6cebcb26b6 100644
--- a/drivers/clk/rockchip/clk-rk3399.c
+++ b/drivers/clk/rockchip/clk-rk3399.c
@@ -829,7 +829,8 @@ static struct rockchip_clk_branch rk3399_clk_branches[] __initdata = {
 			RK3399_CLKGATE_CON(15), 6, GFLAGS),
 
 	/* vcodec */
-	COMPOSITE(0, "aclk_vcodec_pre", mux_pll_src_cpll_gpll_npll_ppll_p, 0,
+	COMPOSITE(ACLK_VCODEC_PRE, "aclk_vcodec_pre",
+			mux_pll_src_cpll_gpll_npll_ppll_p, 0,
 			RK3399_CLKSEL_CON(7), 6, 2, MFLAGS, 0, 5, DFLAGS,
 			RK3399_CLKGATE_CON(4), 0, GFLAGS),
 	COMPOSITE_NOMUX(0, "hclk_vcodec_pre", "aclk_vcodec_pre", 0,
@@ -853,7 +853,7 @@ static struct rockchip_clk_branch rk3399_clk_branches[] __initdata = {
 			RK3399_CLKSEL_CON(9), 14, 2, MFLAGS, 8, 5, DFLAGS,
 			RK3399_CLKGATE_CON(4), 5, GFLAGS),
 
-	COMPOSITE(0, "aclk_vdu_pre", mux_pll_src_cpll_gpll_npll_ppll_p, 0,
+	COMPOSITE(ACLK_VDU_PRE, "aclk_vdu_pre", mux_pll_src_cpll_gpll_npll_ppll_p, 0,
 			RK3399_CLKSEL_CON(8), 6, 2, MFLAGS, 0, 5, DFLAGS,
 			RK3399_CLKGATE_CON(4), 2, GFLAGS),
 	COMPOSITE_NOMUX(0, "hclk_vdu_pre", "aclk_vdu_pre", 0,
diff --git a/include/dt-bindings/clock/rk3399-cru.h b/include/dt-bindings/clock/rk3399-cru.h
index 22cb1dfa9004..dd13554aaf76 100644
--- a/include/dt-bindings/clock/rk3399-cru.h
+++ b/include/dt-bindings/clock/rk3399-cru.h
@@ -231,6 +231,8 @@
 #define ACLK_GIC_PRE	 		262
 #define ACLK_VOP0_PRE	 		263
 #define ACLK_VOP1_PRE	 		264
+#define ACLK_VCODEC_PRE		265
+#define ACLK_VDU_PRE			266
 
 /* pclk gates */
 #define PCLK_PERIHP			320
 
--- a/arch/arm64/boot/dts/rockchip/rk3399.dtsi	2019-10-12 00:42:50.461002500 +0000
+++ b/arch/arm64/boot/dts/rockchip/rk3399.dtsi	2019-10-12 04:46:32.043559737 +0000
@@ -1400,6 +1400,8 @@
 		interrupt-names = "vpu_mmu";
 		clocks = <&cru ACLK_VCODEC>, <&cru HCLK_VCODEC>;
 		clock-names = "aclk", "hclk";
+		assigned-clocks = <&cru ACLK_VCODEC_PRE>;
+       assigned-clock-parents = <&cru PLL_GPLL>;
 		power-domains = <&power RK3399_PD_VCODEC>;
 		#iommu-cells = <0>;
 	};
@@ -1416,6 +1418,10 @@
 			 <&cru SCLK_VDU_CA>, <&cru SCLK_VDU_CORE>;
 		clock-names = "aclk_vcodec", "hclk_vcodec",
 			      "clk_cabac", "clk_core";
+        assigned-clocks = <&cru ACLK_VDU_PRE>, <&cru SCLK_VDU_CA>,
+             <&cru SCLK_VDU_CORE>;
+        assigned-clock-parents = <&cru PLL_NPLL>, <&cru PLL_NPLL>,
+             <&cru PLL_NPLL>;
 		resets = <&cru SRST_H_VDU>, <&cru SRST_A_VDU>,
 			<&cru SRST_VDU_CORE>, <&cru SRST_VDU_CA>,
 			<&cru SRST_A_VDU_NOC>, <&cru SRST_H_VDU_NOC>;

From 3e8454859eaf0768ab99dd09f4bc1abd81732881 Mon Sep 17 00:00:00 2001
From: Elaine Zhang <zhangqing@rock-chips.com>
Date: Wed, 11 Sep 2019 09:21:45 +0800
Subject: [PATCH] clk: rockchip: rk3399: fix up the FRAC_MAX_PRATE limit for i2s and uart

I2S and UART may be application requirements higher than 30M frequency,
So increase the input frequency limit of the frac divider.
IC back-end emulation 800M is safety.

Change-Id: Ida63505b5124799ad8a64d70af4403eae121cbb8
Signed-off-by: Elaine Zhang <zhangqing@rock-chips.com>
---
 drivers/clk/rockchip/clk-rk3399.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/clk/rockchip/clk-rk3399.c b/drivers/clk/rockchip/clk-rk3399.c
index 323d55d50ec6..3c41912c7081 100644
--- a/drivers/clk/rockchip/clk-rk3399.c
+++ b/drivers/clk/rockchip/clk-rk3399.c
@@ -21,8 +21,8 @@
 #include <dt-bindings/clock/rk3399-cru.h>
 #include "clk.h"
 
-#define RK3399_I2S_FRAC_MAX_PRATE       600000000
-#define RK3399_UART_FRAC_MAX_PRATE	600000000
+#define RK3399_I2S_FRAC_MAX_PRATE   800000000
+#define RK3399_UART_FRAC_MAX_PRATE	800000000
 #define RK3399_SPDIF_FRAC_MAX_PRATE	600000000
 #define RK3399_VOP_FRAC_MAX_PRATE	600000000
 #define RK3399_WIFI_FRAC_MAX_PRATE	600000000

